draft-handrews-jri.md

```diff
@@ -0,0 +1,536 @@
+
+By definition, a standalone JRI implementation is an implementation that, in its default configuration, can only correctly process JRI in context-independent documents.
+
+Such implementations MAY offer non-default configurations that incorporate context-aware behavior from any known context specification(s).  Implementations SHOULD document the conditions under which such non-default configurations are safe to use, and any assumptions involved that could produce unexpected behavior if used with a document that does not conform to the relevant context specification.
+
+## The interoperable JRI Subset {#interop-subset}
+
+Context-independent interoperability places several additional requirements on any document that needs to be processed by a context-independent JRI implementation.  Most of these requirements are not enforceable by a context-independent implementation.  Users of context-independent implementations MUST NOT expect documents that violate these requirements to be detected by the implementation, and MUST NOT expect predictable, interoperable behavior
+from processing such documents.
+
+A document that can be interoperably processed by a context-independent JRI implementation will be referred to in this section as a _context-independent document_.  A data format for which all valid documents are context-independent documents will be referred to in this section as a _context-independent format._
+
+The requirements below are stated for context-independent formats.  A document in a non-context-independent format can be context-independent if it avoids using any features forbidden by the context-independent format requirements.
+
+### Restricting base IRI changes and IRI assignment {#interop-overlap}
+
+A context-independent format MUST NOT offer features for assigning IRIs to primary or secondary resources, or changing the base IRI from within a document.
+
+### Recognizing JRI keywords
+
+The property names in the value of `"$defs"` are known to be non-keyword properties as this object syntax is defined by JRI.  These property names therefore MUST be considered exempt from the other requirements in this section regarding property names.
+
+A context-independent format MUST NOT allow object properties with names matching JRI keywords and values that appear valid for that keyword that are not intended to function as JRI keywords anywhere within the format.
+
+Note that as JRI implementations are not required to validate keyword values, simply having the correct value type (e.g. string) is enough to make such a keyword appear to be a valid JRI keyword.  Context-independent JRI implementations MAY offer a non-default configuration option to validate potential JRI keyword values and ignore (rather than treat as an error) those that do not have valid values.
+
+### Ensuring consistent reference semantics
+
+A context-independent format MUST NOT allow any non-JRI-reference keywords in the same object as any JRI reference keyword.[^11]
```

Comment:
Does this mean that `foo` isn't allowed to be a sibling of `$ref`?  How will that work in JSON Schema where we now allow sibling keywords?

Reply:
I see [line 397](https://github.com/json-schema-org/referencing/pull/2/files#diff-a52f2848d61740c55c6f7756031d478fc004b2e4a369b6456e5eec2a1563988dR397) but the wording here seems to contradict that.

Reply:
@gregsdennis JSON Schema is not a context-independent format.